seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/java/org/apache/seatunnel/e2e/connector/fake/FlinkMetricsIT.java

```diff
@@ -0,0 +1,116 @@
+
+    @TestTemplate
+    public void testFlinkMetrics(TestContainer container) throws IOException, InterruptedException {
+        Container.ExecResult executeResult =
+                container.executeJob("/fake_to_assert_verify_flink_metrics.conf");
+        Assertions.assertEquals(0, executeResult.getExitCode());
+        final String jobListUrl = "http://%s:8081/jobs/overview";
+        final String jobDetailsUrl = "http://%s:8081/jobs/%s";
+        final String jobAccumulatorUrl = "http://%s:8081/jobs/%s/vertices/%s/accumulators";
+        final String jobManagerHost;
+        String dockerHost = System.getenv("DOCKER_HOST");
+        if (dockerHost == null) {
+            jobManagerHost = "localhost";
+        } else {
+            URI uri = URI.create(dockerHost);
+            jobManagerHost = uri.getHost();
+        }
+        // create http client
+        CloseableHttpClient httpClient = HttpClients.createDefault();
+
+        // get job id
+        HttpGet httpGet = new HttpGet(String.format(jobListUrl, jobManagerHost));
+        CloseableHttpResponse response = httpClient.execute(httpGet);
+        Assertions.assertEquals(response.getStatusLine().getStatusCode(), 200);
+        String responseContent = EntityUtils.toString(response.getEntity());
+        ObjectNode jsonNode = JsonUtils.parseObject(responseContent);
+        String jobId = jsonNode.get("jobs").get(0).get("jid").asText();
+        Assertions.assertNotNull(jobId);
+
+        // get job vertices
+        httpGet = new HttpGet(String.format(jobDetailsUrl, jobManagerHost, jobId));
+        response = httpClient.execute(httpGet);
+        Assertions.assertEquals(response.getStatusLine().getStatusCode(), 200);
+
+        responseContent = EntityUtils.toString(response.getEntity());
+        jsonNode = JsonUtils.parseObject(responseContent);
+        String verticeId = jsonNode.get("vertices").get(0).get("id").asText();
+
+        // get metrics
+        httpGet = new HttpGet(String.format(jobAccumulatorUrl, jobManagerHost, jobId, verticeId));
+        response = httpClient.execute(httpGet);
+        responseContent = EntityUtils.toString(response.getEntity());
+        jsonNode = JsonUtils.parseObject(responseContent);
+        JsonNode metrics = jsonNode.get("user-accumulators");
+
+        int size = metrics.size();
+
+        for (int i = 0; i < size; i++) {
+            JsonNode metric = metrics.get(i);
+            String name = metric.get("name").asText();
+            String value = metric.get("value").asText();
+            LOGGER.info("Accumulator [{}]'s value = [{}]", name, value);
+            Assertions.assertNotNull(value);
+            switch (name) {
+                case MetricNames.SOURCE_RECEIVED_COUNT:
+                case MetricNames.SINK_WRITE_COUNT:
+                    Assertions.assertEquals(5, Integer.valueOf(value));
+                    break;
+                case MetricNames.SOURCE_RECEIVED_BYTES:
```

Comment:
Move the inspection outside the for loop

e.g.

for () {
...
}
assert SOURCE_RECEIVED_COUNT = 5
assert SINK_WRITE_COUNT = 5
assert SOURCE_RECEIVED_BYTES = 2160
assert SINK_WRITE_BYTES = 2160

Reply:
What' the meaning of this?

Reply:
Make sure all 4 indicators exist and have correct values

Reply:
For example, this test cannot detect SINK_WRITE_COUNT after it is deleted.

Reply:
> Make sure all 4 indicators exist and have correct values

Flink1.13 can not record writer accumulator restricted by code limitations. The case you're talking about is extremely extreme, it's impossible for it to happen.

Reply:
It is a very common problem that functionality is lost due to subsequent contributor modifications.

Annotations can be used to skip versions that do not require testing
```java
@DisabledOnContainer(
         value = {TestContainerId.FLINK_1_13},
         type = {EngineType.SPARK, EngineType.SEATUNNEL})
public class FlinkMetricsIT extends TestSuiteBase {
```

And we tell users to only support versions above flink1.13

Reply:
Done, please review.